src/core/lib/iomgr/udp_server.c

```diff
@@ -217,14 +217,23 @@ static int prepare_socket(int fd, const struct sockaddr *addr,
     goto error;
   }
 
-  if (!grpc_set_socket_nonblocking(fd, 1) || !grpc_set_socket_cloexec(fd, 1)) {
-    gpr_log(GPR_ERROR, "Unable to configure socket %d: %s", fd,
-            strerror(errno));
+  if (grpc_set_socket_nonblocking(fd, 1) != GRPC_ERROR_NONE) {
+    gpr_log(GPR_ERROR, "Unable to set nonblocking %d: %s", fd, strerror(errno));
```

Comment:
@dgquintas thanks, these changes are intended to switch over to use grpc_error as simply as possible. I am working on a followup that will change udp_server.c to be more similar to tcp_server_posix.c in terms of error handling.

Also this is a step to fixing https://github.com/grpc/grpc/issues/6624 ... but this stuff needs to build first.